### PR TITLE
Add Jekyll Plugin instead of Gulp collection-creating step

### PR DESCRIPTION
**Pull request by tdooner**
I really liked the cleanliness of the Gulp processing code, but it feels like a shame to introduce another step into this already-complicated build process for the site. We looked at the documentation for Jekyll plugins at OO the other night and I thought it would be worth an attempt to see what it looks like.

I wasn't able to factor the code as nicely as I wanted (e.g. the `#from_json` method). But, this works pretty well. It uses a Generator to add Jekyll::Page objects that are auto-created based on the JSON files in the API directory.

Using some other plugin trickery, we can expose these pages in the Liquid template context as well -- so constructs like `site.candidates` still work!

As I said in one of the commit messages, the main downside to this plugin-based approach is that we don't get the auto-compilation from Github (as executing arbitrary Ruby is disabled for obvious reasons). We will have to set up travis to do it. In fact, maybe we can just merge this in with the other repo.

### CODE REVIEW

File: _layouts/ballot.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+layout: default
+---
+{% assign ballot = page %}
```
Comment (adborden) [inline _layouts/ballot.html]: 👏 I was thinking the same thing!

**Comment by adborden** [comment]
How do you see the repos getting merged? Would we move the campaign finance calculators into the plugin?

I kind of like that things like the ballot, locality, candidates, and referendums were plain text yaml. I _think_ that it would allow other folks to easily contribute to elections in new localities without having to write code. We'd split the finance data into separate generated files. The jury is still out on whether this would actually work better than the google sheet.

**Comment by tdooner** [comment]
r.e. merging- If there are two repos that both run ruby code to build static files, it seems natural to me to combine them into one if possible. I know that this will become javascript heavy, but still. I was imagining namespacing the backend files in a directory somewhere and then wiring it all up in the makefile.

r.e. YAML and editability of ballot/locality/candidates/referendums - I could be convinced we should use YAML instead, though I think our spreadsheets right now are pretty good from a user-friendliness angle (i.e. you don't need to know git to change it.) Either way, there should just be one way to do it, and I leaned towards repurposing the backend structure since the data is already processed there.

**Comment by adborden** [comment]
> r.e. merging- If there are two repos that both run ruby code to build static files, it seems natural to me to combine them into one if possible.

Yep, I completely agree, sorry if that wasn't clear. I was just interested in how you saw that happening.

> I know that this will become javascript heavy, but still. I was imagining namespacing the backend files in a directory somewhere and then wiring it all up in the makefile.

You mean Ruby-heavy? The part that I'm curious about is why having the jekyll plugin _and_ the Ruby Calculators. Does it make sense to combine them?

I was thinking that since backend-static is already generating static data files, why not just have it generate those files into the correct Jekyll Collection directories and skip the plugin? Or, are you thinking of moving the Calculators into the plugin?

> r.e. YAML and editability of ballot/locality/candidates/referendums - I could be convinced we should use YAML instead, though I think our spreadsheets right now are pretty good from a user-friendliness angle (i.e. you don't need to know git to change it.) Either way, there should just be one way to do it, and I leaned towards repurposing the backend structure since the data is already processed there.

For the yaml, I was thinking folks could just edit directly on GitHub or maybe http://prose.io/ which is a slightly nicer interface  (I don't know how it handles yaml, though). No git necessary just a web interface.

This might be a separate discussion, but I would like to kill the google sheet. The problem for me is that that it's not easy to ask folks to add new localities to it. You have to invite/approve their access to the sheet. If folks prefer working in the sheet over yaml, then I would prefer to setup a template Google Sheet that folks could copy, fill out, and then export to a CSV which they could then add to the repo on Github.

**Comment by tdooner** [comment]
> This might be a separate discussion, but I would like to kill the google sheet.
Yes! Definitely.

The order of operations I'm seeing here, and I'm curious how you'd tweak it, is:

1. Get this repo to an MVP status (with tolerable navigation; all numbers are displaying as on existing site; no ODCA styling or graphs)
2. Begin work to theme this repo to look like the existing ODCA. (without javascript, so maybe the graphs will be static and the tables won't sort)
3. At this point we could probably deploy it to production in place of the existing ODCA
4. Add a javascript build pipeline, and begin to re-add the interactivity
5. Simultaneously, we can work to streamline the build process
  * Remove the spreadsheet in favor of auto-detecting campaign committee names somehow
  * Remove intermediate JSON files / the idea of the API, or at least bake them into the Jekyll build so that the static API doesn't require a sintara web server.
6. New features, like Search.

**Comment by adborden** [comment]
Great, I added these as cards to a temporary [MVP project board](https://github.com/adborden/odca-jekyll/projects/1) so we can sort them. I elaborated on some and tried to make them a little more actionable/focused.

Question: If we lose the ability to publish to GH pages with this PR, shouldn't we do some CI/CD work up front?

I moved the "navigation" card to "done", since I think the existing nav we have is done, minus a few pages which are captured in other cards.

For the non-MVP cards, I'm going to create them in backend-static.

**Comment by adborden** [comment]
ah, one more... At what point do you think we can merge the repos? I think this could happen immediately, but there's not necessarily a reason to do it except to keep everything in a single repo.

**Comment by adborden** [comment]
I'm thinking we should hold off on merging this and think about creating a plugin as part of merging this with backend-static and maybe push that off until later.

Main reason is that I like this repo is standalone, but the jekyll plugin ties it to backend-static. If we're just working with yaml files, it's still really easy to change the ballot-side of things and we can focus on the ballot models, their presentation, information architecture, and navigation.

**Pull request closed without merging**